tools/crm_resource_runtime.c

```diff
@@ -367,42 +379,53 @@ cli_resource_delete_attribute(resource_t *rsc, const char *requested_name,
     }
 
     if(safe_str_eq(attr_set_type, XML_TAG_META_SETS)) {
-        rsc = find_matching_attr_resource(rsc, requested_name, attr_set,
+        resources = find_matching_attr_resources(rsc, requested_name, attr_set,
                                           attr_id, attr_name, cib, "delete");
+    } else {
+        resources = g_list_append(resources, rsc);
     }
 
-    lookup_id = clone_strip(rsc->id);
-    rc = find_resource_attr(cib, XML_ATTR_ID, lookup_id, attr_set_type, attr_set, attr_id, attr_name,
-                            &local_attr_id);
+    for(GList *gIter = resources; gIter; gIter = gIter->next) {
+        char *lookup_id = NULL;
+        xmlNode *xml_obj = NULL;
+        char *local_attr_id = NULL;
 
-    if (rc == -ENXIO) {
-        free(lookup_id);
-        return pcmk_ok;
+        rsc = (resource_t*)gIter->data;
 
-    } else if (rc != pcmk_ok) {
-        free(lookup_id);
-        return rc;
-    }
+        lookup_id = clone_strip(rsc->id);
+        rc = find_resource_attr(cib, XML_ATTR_ID, lookup_id, attr_set_type, attr_set, attr_id, attr_name,
+                                &local_attr_id);
 
-    if (attr_id == NULL) {
-        attr_id = local_attr_id;
-    }
+        if (rc == -ENXIO) {
+            free(lookup_id);
+            return pcmk_ok;
```

Comment:
-ENXIO means the attribute was not found, which means nothing has to be done for delete.

However returning immediately makes sense only when there is one resource, for the list you'll have to handle it differently.

Reply:
Fixed. resources list is always destroyed before leaving either `cli_resource_delete_attribute` or `cli_resource_update_attribute` functions.

Reply:
If there are multiple resources in the list, and the attribute already doesn't exist for one of them, processing will stop there, and none of the later resources in the list will have the attribute deleted.

Reply:
Aha, I got the point. I think it's fixed now.